src/Tests/CoreTests/Geometry/GeometryTests.hpp

```diff
@@ -17,34 +17,34 @@ namespace RaTests
     {
         void run() override
         {
-            Vector3 a(1.f, 2.3f, 4.5f);
-            Vector3 b(-6.f, 7.f, 8.9f);
-            Vector3 c(-3.f, 12.3f, -42.1f);
+            Vector3 a(1., 2.3, 4.5);
+            Vector3 b(-6., 7., 8.9);
+            Vector3 c(-3., 12.3, -42.1);
 
             // Midpoint.
-            Vector3 m = 0.5f*(a+b);
+            Vector3 m = 0.5*(a+b);
 
             // Point on the line, before A
-            Vector3 na = a - 12.f*(b-a);
+            Vector3 na = a - 12.*(b-a);
 
             // Point on the line after B
-            Vector3 nb = a + 42.f*(b-a);
+            Vector3 nb = a + 42.*(b-a);
 
 
             Vector3 y,z;
             Ra::Core::Vector::getOrthogonalVectors(b-a, y, z);
 
             // Test line queries.
-            RA_UNIT_TEST(  pointToLineSq(a, a, b-a) == 0.f, "distance from A to AB" );
-            RA_UNIT_TEST(  pointToLineSq(b, a, b-a) == 0.f, "distance from B to AB" );
-            RA_UNIT_TEST(  areApproxEqual(pointToLineSq(na, a, b-a), 0.f), "point on the line.");
-            RA_UNIT_TEST(  areApproxEqual(pointToLineSq(nb, a, b-a), 0.f), "point on the line.");
+            RA_UNIT_TEST(  areApproxEqual(pointToLineSq(a, a, b-a), Scalar(0.)), "distance from A to AB" );
```

Comment:
What is the rationale behind these changes ? I think it makes sense to ask for the distance from A to B to be *exactly* 0. Were the tests failing on your machine ?

Reply:
nope, actually I don't care ... just one was change to use areApproxEqual, the other not.
Since there is no special case in pointToLineSq that test extremities, and float computation could have roundup errors, I think it is better to keep areApproxEqual

Reply:
Well, it's more about our *specification* of what the result of the algorithm should be, because this unit test is implicitly a specification of the algorithm.

Do we want to make sure that the distance between a segment and one of its extremities is always *exaclty* equal to zero ? I think it makes sense (just like the distance from a point to itself should be exactly zero). If we decide this, the implementation *has* to make it so that the result cannot suffer from roundup errors (which is the case for the current implementation).

Reply:
from the code of pointToLineSq
                `return (dir.cross(q - a)).squaredNorm() / dir.squaredNorm();`
so `a` must return 0, `b` don't.
Is it more readable to have always approxEqual ? don't know. I let you choose and commit